javascript/src/jobProcessor.ts

```diff
@@ -157,13 +159,18 @@ export const jobProcessor = async (opts?: ConnectionOptions) => {
           clearInterval(extendAckTimer)
         }
         deferred.done()
+        deferreds.pop()
       }
       // Don't process any more messages if stopping
       if (abortController.signal.aborted) {
         return
       }
     }
   }
+  /**
+   * To be used from caller to await all jobs to be completed
+   */
+  const finish = () => Promise.all(_.map('promise', deferreds))
```

Comment:
Why not wait for the jobs to complete in `stop`?

Reply:
> Why not wait for the jobs to complete in `stop`? After all, we're passing them a `signal` so they have the opportunity to stop quickly.

Well, the idea really is that you might want to just block until everything completes but not really stop processing and abort.

Reply:
Do you have an example where this would be useful?